src/System.Diagnostics.Tracing.Logger/src/System/Diagnostics/Tracing/Logger.cs

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace System.Diagnostics.Tracing
+{
+    /// <summary>
+    /// Logger is meant for diagnostic logging.     It is meant to be reasonably familiar to both users of other
+    /// common logging APIs (e.g. NLOG, Log4Net etc), and act at as a front-end facade for them.   In particular 
+    /// it has the concept of a LogLevel (verbosity) and Logger names (thus you can have a logger per component).   
+    /// 
+    /// System.Diagnostics.Tracing.Logger serves much the same purpose as System.Diagnostics.Tracing.EventSource.   
+    /// In general the guidance is to use EventSource if you have the choice.   There are two main reasons why
+    /// you may wish to use Logger rather than EventSource 
+    /// 
+    ///   1) You need multi-tenancy.   There is a single 'hub' in a given appdomain/process for all EventSources 
+    ///      in that appdomain/process.   Thus there can only be one EventSource call 'MyProgram'.   This is a
+    ///      problem if like ASP.NET, which wishes to support many independent tenants in the same appdomain.  
+    ///      Because the events form one tenant will mix with the others.  
+    ///   2) Dependency Injection - Logger supports ILogger, which allows you to build your application so that
+    ///      the exact implemenation of the logging system can be passed into it.   Logger has support for this.
+    /// 
+    /// If you don't have these requirements you should seriously consider using EventSource instead.   However
+    /// if you are using ASP.NET 5, they support both (1) and (2) above, and it is best to stick with Logger.  
+    /// 
+    /// Logger leverages DiagnosticListener as its 'plumbing'.   Thus Every Logger is a DiagnosticListener which  
+    /// means that you can discover them using DiagnosticListener.AllListeners.   Once you have the Logger you want, 
+    /// you can use the Subscribe() method to connect to the stream of data.  
+    /// 
+    /// System.Diagnostics.Tracing.EventSource will have plumbing to treat all DiagnosticSource (and thus all 
+    /// Loggers), as EventSources (as long as the payloads are serializable).   Thus it is also possible access
+    /// the data stream of Loggers from an EventListener and therefore ETW and LTTng.  
+    /// </summary>
+    public class Logger : DiagnosticListener, ILogger
```

Comment:
It seems strange that a Logger is a DiagnosticListener... since the purpose of logging is writing to it, wouldn't it be a DiagnosticSource that others listen to, just as EventSources write and are listened to by EventListeners?